frontend/packages/knative-plugin/src/components/knatify/CreateKnatifyPage.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik, FormikHelpers } from 'formik';
+import { RouteComponentProps } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { Helmet } from 'react-helmet';
+import { useActivePerspective } from '@console/shared';
+import { useExtensions, Perspective, isPerspective } from '@console/plugin-sdk';
+import { ProjectModel } from '@console/internal/models';
+import {
+  LoadingBox,
+  Firehose,
+  FirehoseResource,
+  history,
+} from '@console/internal/components/utils';
+import NamespacedPage, {
+  NamespacedPageVariants,
+} from '@console/dev-console/src/components/NamespacedPage';
+import DeployImageForm from '@console/dev-console/src/components/import/DeployImageForm';
+import { handleRedirect } from '@console/dev-console/src/components/import/import-submit-utils';
+import { DeployImageFormData } from '@console/dev-console/src/components/import/import-types';
+import { deployValidationSchema } from '@console/dev-console/src/components/import/deployImage-validation-utils';
+import {
+  getInitialValuesKnatify,
+  knatifyResources,
+  getKnatifyWorkloadData,
+} from '../../utils/knatify-utils';
+
+const CreateKnatifyComponentLoader: React.FunctionComponent<any> = (props: any) => {
+  const { loaded, appName, namespace, handleSubmit, projects, workloadResource } = props;
+  const { t } = useTranslation();
+  return loaded && !_.isEmpty(workloadResource.data) ? (
+    <Formik
+      initialValues={getInitialValuesKnatify(
+        getKnatifyWorkloadData(workloadResource.data),
+        appName,
+        namespace,
+      )}
+      validationSchema={deployValidationSchema(t)}
+      onSubmit={handleSubmit}
+      onReset={history.goBack}
+    >
+      {(formikProps) => <DeployImageForm {...formikProps} projects={projects} isKnatify />}
+    </Formik>
+  ) : (
+    <LoadingBox />
+  );
+};
+
+type CreateKnatifyPageProps = RouteComponentProps<{ ns?: string }>;
+
+const CreateKnatifyPage: React.FunctionComponent<CreateKnatifyPageProps> = ({
+  match,
+  location,
+}) => {
+  const { t } = useTranslation();
+  const namespace = match.params.ns;
+  const queryParams = new URLSearchParams(location.search);
+  const kind = queryParams.get('kind');
+  const appName = queryParams.get('name');
```

Comment:
Since we can go to the page through route. if I enter an invalid kind or name or namespace in the route console breaks with error or similar to this one
<img width="616" alt="Screenshot 2021-02-23 at 2 59 10 PM" src="https://user-images.githubusercontent.com/9278015/108824991-6f979680-75e8-11eb-8646-426c248e0055.png">
I think we should have safe checks here. WDYT?

Reply:
updated